### ISSUE TITLE
Add update functionality to edge gateway

### ISSUE THREAD

**Issue by dataclouder** [vmware/go-vcloud-director#202]
As defined in PR #195, we support edge gateway creation and deletion.

To complete the set of functionalities, we should also support updates.

While the [API for updates](https://code.vmware.com/apis/553/vcloud-director/doc/doc/operations/PUT-EdgeGateway.html) is conceptually simple, as it requires passing the same structure used for creation, determining what can be updated and how is more complicated.

For example, when we change the dependency from an external network as default gateway, we need to change the elements that we know (gateway IP and netmask) but also elements that we don't know directly (the external IP allocated or whether there is an available IP). Also, changing the external network will influence how the depending routed networks will respond.

Moreover, we need to understand which parts of the configuration can be altered without side effects (or with manageable side effects) and which should be considered non-alterable, meaning that we will need to re-deploy the edge gateway to change them.

Given the complexity of the matter, we need to decide between two approaches:

*. create an `Update` function, with a structure similar to the one used for `CreateEdgeGateway` and handle all the possibilities in there.
* create a separate function for each element that can be changed (such as `UnlinkNetwork`, `RemoveDefaultGateway`, `ChangeDefaultGateway`, and so on)

**Comment by lvirbalas** [comment]
I agree about the complexity and as such like the second suggestion more (with separate functions foreach updatable element). This way we can roll out updates incrementally and manage risks better.

**Comment by lvirbalas** [comment]
@slavaavr in the light of your recent PR #287 I hope you'll be interested in trying out this PR when it's ready for merge?

**Comment by bogi0704** [comment]
I would like to use the update functionality. Does anyone have an update about this?
